### PR TITLE
feat: add `blas/base/drotm`

### PR DESCRIPTION
**Pull request by aman-095**
## Description

> What is the purpose of this pull request?

This RFC proposes to add a routine to apply a modified Givens plane rotation on double-precision floating-point vectors as defined in [BLAS Level 1](http://www.netlib.org/blas/#_level_1) routines. Specifically adding `@stdlib/blas/base/drotm` is proposed.

## Related Issues

> Does this pull request have any related issues?

None. 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/base/drotm/README.md [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+
+-   **N**: `[in] CBLAS_INT` number of indexed elements.
+-   **X**: `[inout] double*` first input array.
+-   **strideX**: `[in] CBLAS_INT` index increment for `X`.
+-   **Y**: `[inout] double*` first input array.
+-   **strideY**: `[in] CBLAS_INT` index increment for `Y`.
+-   **param**: `[in] double` parameters for the modified Givens transformation.
+
+```c
+void c_drotm( const CBLAS_INT N, double *X, const CBLAS_INT strideX, double *Y, const CBLAS_INT strideY, const double param[ 5 ] );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/base/drotm/README.md]:
```suggestion
void c_drotm( const CBLAS_INT N, double *X, const CBLAS_INT strideX, double *Y, const CBLAS_INT strideY, const double *param);
```

File: lib/node_modules/@stdlib/blas/base/drotm/benchmark/fortran/benchmark.length.f [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    do i = 1, len
+      call random_number( r )
+      x( i ) = ( r*200.0d0 ) - 100.0d0
+      y( i ) = ( r*200.0d0 ) - 100.0d0
+    end do
+    param(1) = 0.0d0
+    param(2) = 0.0d0
+    param(3) = 0.0d0
+    param(4) = 0.0d0
+    param(5) = 0.0d0
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/base/drotm/benchmark/fortran/benchmark.length.f:153]:
```suggestion
    param = 0.0d0
```

Try this, I think this should work.

File: lib/node_modules/@stdlib/blas/base/drotm/examples/c/example.c [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    // Specify parameters for the modified Givens transformation:
+    const double param[5] = { 0.0, 0.0, 2.0, -3.0, 0.0 };
+
+    // Apply plane rotation:
+    c_drotm( N, x, strideX, y, strideY, param );
+
+    // Print the result:
+    for ( int i = 0; i < 5; i++ ) {
+        printf( "x[ %i ] = %lf, y[ %i ] = %lf\n", i, x[ i ], i, y[ i ] );
+    }
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/base/drotm/examples/c/example.c]: tab indentation

File: lib/node_modules/@stdlib/blas/base/drotm/include/stdlib/blas/base/drotm.h [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+* If C++, prevent name mangling so that the compiler emits a binary file having undecorated names, thus mirroring the behavior of a C compiler.
+*/
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+* Applies a modified Givens plane rotation.
+*/
+void API_SUFFIX(c_drotm)( const CBLAS_INT N, double *X, const CBLAS_INT strideX, double *Y, const CBLAS_INT strideY, const double param[ 5 ] );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/base/drotm/include/stdlib/blas/base/drotm.h]:
```suggestion
void API_SUFFIX(c_drotm)( const CBLAS_INT N, double *X, const CBLAS_INT strideX, double *Y, const CBLAS_INT strideY, const double *param );
```

File: lib/node_modules/@stdlib/blas/base/drotm/include/stdlib/blas/base/drotm_cblas.h [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+* If C++, prevent name mangling so that the compiler emits a binary file having undecorated names, thus mirroring the behavior of a C compiler.
+*/
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+* Applies a modified Givens plane rotation.
+*/
+void API_SUFFIX(cblas_drotm)( const CBLAS_INT N, double *X, const CBLAS_INT strideX, double *Y, const CBLAS_INT strideY, const double param[ 5 ] );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/base/drotm/include/stdlib/blas/base/drotm_cblas.h]: same comment here

File: lib/node_modules/@stdlib/blas/base/drotm/lib/drotm.js [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+	var ix;
+	var iy;
+	var i;
+	var w;
+	var z;
+
+	ix = 0;
+	iy = 0;
+	dflag = param[ 0 ];
+	if ( N <= 0 || ( dflag+2.0 ) === 0.0 ) {
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/base/drotm/lib/drotm.js]:
```suggestion
	if ( N <= 0 || ( dflag + 2.0 ) === 0.0 ) {
```

File: lib/node_modules/@stdlib/blas/base/drotm/lib/ndarray.js [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+	var ix;
+	var iy;
+	var i;
+	var w;
+	var z;
+
+	ix = offsetX;
+	iy = offsetY;
+	dflag = param[ 0 ];
+	if ( N <= 0 || ( dflag+2.0 ) === 0.0 ) {
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/base/drotm/lib/ndarray.js]:
```suggestion
	if ( N <= 0 || ( dflag + 2.0 ) === 0.0 ) {
```

File: lib/node_modules/@stdlib/blas/base/drotm/src/drotm.c [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+	double dh21;
+	double dh22;
+	CBLAS_INT i;
+	double w;
+	double z;
+
+	ix = 0;
+	iy = 0;
+	dflag = param[ 0 ];
+	if ( N <= 0 || ( dflag+2.0 ) == 0.0 ) {
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/base/drotm/src/drotm.c]:
```suggestion
	if ( N <= 0 || ( dflag + 2.0 ) == 0.0 ) {
```

File: lib/node_modules/@stdlib/blas/base/drotm/src/drotm.f [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+  ! ..
+  ! Local scalars:
+  real(dp) :: dflag, dh11, dh12, dh21, dh22, two, w, z, zero
+  integer :: i, kx, ky, nsteps
+  ! ..
+  ! Data statements:
+  data zero / 0.0d0 /, two / 2.0d0 /
+  ! ..
+  dflag = dparam(1)
+  if( n <= 0 .OR. (dflag+two == zero)) return
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/base/drotm/src/drotm.f]:
```suggestion
  if( n <= 0 .OR. ( dflag + two == zero ) ) return
```

File: lib/node_modules/@stdlib/blas/base/drotm/src/drotm.f [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+  if(strideX == strideY .AND. strideX > 0) then
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/base/drotm/src/drotm.f]:
```suggestion
  if( strideX == strideY .AND. strideX > 0 ) then
```

File: lib/node_modules/@stdlib/blas/base/drotm/src/drotm.f [hunk 11]
```diff
@@ -0,0 +1,153 @@
...
+        z = dy(ky)
+        dx(kx) = dh11*w + z
+        dy(ky) = -w + dh22*z
+        kx = kx + strideX
+        ky = ky + strideY
+      end do
+    end if
+  end if
+  return
+end subroutine drotm
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/base/drotm/src/drotm.f:166]: fix spacing issue in this file.

File: lib/node_modules/@stdlib/blas/base/drotm/src/drotm_cblas.c [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+* Applies a modified Givens plane rotation.
+*
+* @param N        number of indexed elements
+* @param X        input array
+* @param strideX  X stride length
+* @param Y        output array
+* @param strideY  Y stride length
+* @param param    parameters for the modified Givens transformation
+*/
+void API_SUFFIX(c_drotm)( const CBLAS_INT N, double *X, const CBLAS_INT strideX, double *Y, const CBLAS_INT strideY, const double param[ 5 ] ) {
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/base/drotm/src/drotm_cblas.c]: change API definition accordingly

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
Minor changes required, then this PR can be reviewed again.

**Comment by kgryte** [comment]
@aman-095 LMK once you've update this PR to match the changes in https://github.com/stdlib-js/stdlib/pull/2267.

**Comment by aman-095** [comment]
Hey, @kgryte, I have applied the necessary changes to this PR.

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @aman-095!

**Pull request merged**